File: snippets/fsharp/System/Double/TryParse/tryparse2.fs
module tryparse2

// <Snippet3>
open System

[<EntryPoint>]
let main _ = 
    let value = string Double.MinValue
    match Double.TryParse value with
    | true, number ->
        printfn $"{number}"
    | _ ->
        printfn $"{value} is outside the range of a Double."

    let value = string Double.MaxValue
    match Double.TryParse value with
    | true, number ->
        printfn $"{number}"
    | _ ->
        printfn $"{value} is outside the range of a Double."

    0
// The example displays the following output:
//    -1.79769313486232E+308 is outside the range of the Double type.
//    1.79769313486232E+308 is outside the range of the Double type.
// </Snippet3>

File: snippets/csharp/System/Double/TryParse/tryparse2.cs
// <Snippet3>
using System;

public class Example
{
    public static void Main()
    {
        string value;
        double number;

        value = double.MinValue.ToString();
        if (double.TryParse(value, out number))
            Console.WriteLine(number);
        else
            Console.WriteLine($"{value} is outside the range of a Double.");

        value = double.MaxValue.ToString();
        if (double.TryParse(value, out number))
            Console.WriteLine(number);
        else
            Console.WriteLine($"{value} is outside the range of a Double.");
    }
}
// The example displays the following output:
//    -1.79769313486232E+308 is outside the range of the Double type.
//    1.79769313486232E+308 is outside the range of the Double type.
// </Snippet3>
